src/main/java/frc/trigon/robot/motorsimulation/MotorSimulation.java

```diff
@@ -0,0 +1,145 @@
+package frc.trigon.robot.motorsimulation;
+
+import com.ctre.phoenix6.controls.DutyCycleOut;
+import com.ctre.phoenix6.controls.MotionMagicVoltage;
+import com.ctre.phoenix6.controls.PositionVoltage;
+import com.ctre.phoenix6.controls.VoltageOut;
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.Notifier;
+import edu.wpi.first.wpilibj.Timer;
+import frc.trigon.robot.constants.RobotConstants;
+import frc.trigon.robot.utilities.Conversions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A class that simulates a motor and its PID and feedforward.
+ */
```

Comment:
A wrapper class for the WPIlib default simulation classes, that'll act similarly to how the TalonFX motor controller works etc.